fe/fe-core/src/main/java/com/starrocks/catalog/IcebergTable.java

```diff
@@ -0,0 +1,292 @@
+import org.apache.iceberg.types.Types;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class IcebergTable extends Table {
+    private static final Logger LOG = LogManager.getLogger(IcebergTable.class);
+
+    private static final String PROPERTY_MISSING_MSG =
+            "Iceberg %s is null. Please add properties('%s'='xxx') when create table";
+    private static final String JSON_KEY_ICEBERG_DB = "database";
+    private static final String JSON_KEY_ICEBERG_TABLE = "table";
+    private static final String JSON_KEY_RESOURCE_NAME = "resource";
+    private static final String JSON_KEY_ICEBERG_PROPERTIES = "icebergProperties";
+
+    private static final String ICEBERG_CATALOG = "starrocks.catalog-type";
+    private static final String ICEBERG_METASTORE_URIS = "iceberg.catalog.hive.metastore.uris";
+    private static final String ICEBERG_DB = "database";
+    private static final String ICEBERG_TABLE = "table";
+    private static final String ICEBERG_RESOURCE = "resource";
+
+    private org.apache.iceberg.Table icebergTable; // actual iceberg table
+    private Optional<Snapshot> snapshot;
+    private TableScan tableScan;
+
+    private String db;
+    private String table;
+    private String resourceName;
+    private String tableLocation;
+
+    private List<String> columnNames = Lists.newArrayList();
+
+    private Map<String, String> icebergProperties = Maps.newHashMap();
+
+    public IcebergTable() {
+        super(TableType.ICEBERG);
+    }
+
+    public IcebergTable(long id, String name, List<Column> schema, Map<String, String> properties) throws DdlException {
+        super(id, name, TableType.ICEBERG, schema);
+        validate(properties);
+    }
+
+    public String getDb() {
+        return db;
+    }
+
+    public String getTable() {
+        return table;
+    }
+
+    public String getResourceName() {
+        return resourceName;
+    }
+
+    public IcebergCatalogType getCatalogType() {
+        return IcebergCatalogType.valueOf(icebergProperties.get(ICEBERG_CATALOG));
+    }
+
+    public String getIcebergHiveMetastoreUris() {
+        return icebergProperties.get(ICEBERG_METASTORE_URIS);
+    }
+
+    public void setTableLocation(String location) {
+        this.tableLocation = location;
+    }
+
+    public TableScan getTableScan(List<UnboundPredicate> icebergPredicates) {
+        initTableScanIfNecessary();
+        icebergPredicates.forEach(predicate -> this.tableScan = tableScan.filter(predicate));
+        return tableScan;
+    }
+
+    public Optional<Snapshot> getSnapshot() {
+        initTableScanIfNecessary();
+        return snapshot;
+    }
+
+    private void initTableScanIfNecessary() {
+        if (this.tableScan == null) {
```

Comment:
Whether the table should be cached, instead of tableScan. 
I think in other follow-up feature, icebergTable may be accessed frequently.
the table can get the latest through refresh.